### PR TITLE
[fix] Fix publishing warnings

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, no (Scala) warnings would be reported since they would come inside stderr in a progress build event. Now, they are handled via that event and cleared correctly when needed.

Are there any diagnostics tests in integration tests anywhere that I could use as examples?

### CODE REVIEW

**Comment by agluszak** [comment]
> Are there any diagnostics tests in integration tests anywhere that I could use as examples?

sadly - no :( but I'll gladly help you design some

**Comment by tgodzik** [comment]
> > Are there any diagnostics tests in integration tests anywhere that I could use as examples?
> 
> sadly - no :( but I'll gladly help you design some

Here https://github.com/build-server-protocol/bsp-testkit2 right? That's the testkit being used in current repository? It seems that it's released manually if I am not mistaken

**Comment by agluszak** [comment]
> That's the testkit being used in current repository?
yes

> It seems that it's released manually if I am not mistaken
I don't think we've ever made a release to maven. Here it is depended upon as a bazel module: https://github.com/JetBrains/bazel-bsp/blob/master/MODULE.bazel#L29

File: server/src/main/kotlin/org/jetbrains/bsp/bazel/server/diagnostics/DiagnosticsService.kt [hunk 1]
```diff
@@ -17,23 +15,31 @@ class DiagnosticsService(workspaceRoot: Path, private val hasAnyProblems: Mutabl
 
     private val parser = DiagnosticsParser()
     private val mapper = DiagnosticBspMapper(workspaceRoot)
+    /* Warnings are reported before the target completed event, when everything is cleared. so we want to avoid removing them */
+    private val updatedInThisRun = mutableSetOf<PublishDiagnosticsParams>()
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply server/src/main/kotlin/org/jetbrains/bsp/bazel/server/diagnostics/DiagnosticsService.kt:19]: Yep, that's a bit terrible 😔 Not sure if I can make it better, but will try.

Any ideas on improving this?

[the remark this answers is not in the document]
Comment (agluszak) [reply server/src/main/kotlin/org/jetbrains/bsp/bazel/server/diagnostics/DiagnosticsService.kt:19]: I guess we have to change the protocol

[the remark this answers is not in the document]
Comment (agluszak) [reply server/src/main/kotlin/org/jetbrains/bsp/bazel/server/diagnostics/DiagnosticsService.kt:19]: for now it can stay like this I think

**Comment by tgodzik** [comment]
> > Are there any diagnostics tests in integration tests anywhere that I could use as examples?
> 
> sadly - no :( but I'll gladly help you design some

Should be enough to do https://github.com/build-server-protocol/bsp-testkit2/pull/45, what do you think?

**Comment by agluszak** [comment]
Hey, any update on this?

In order to add a test you can add a new scala target [here](https://github.com/JetBrains/bazel-bsp/tree/master/e2e/test-resources/sample-repo/scala_targets) and add a test [here](https://github.com/JetBrains/bazel-bsp/blob/master/e2e/src/main/kotlin/org/jetbrains/bsp/bazel/BazelBspScalaProjectTest.kt).

File: e2e/src/main/kotlin/org/jetbrains/bsp/bazel/BazelBspScalaProjectTest.kt [hunk 2]
```diff
@@ -19,44 +14,49 @@ object BazelBspScalaProjectTest : BazelBspTestBaseScenario() {
...
-      ScalacOptionsItem(
-        it,
-        emptyList(),
-        listOf(
-          "file://\$BAZEL_OUTPUT_BASE_PATH/execroot/__main__/bazel-out/k8-fastbuild/bin/external/io_bazel_rules_scala_scala_library/io_bazel_rules_scala_scala_library.stamp/scala-library-2.13.6-stamped.jar",
-          "file://\$BAZEL_OUTPUT_BASE_PATH/execroot/__main__/bazel-out/k8-fastbuild/bin/external/io_bazel_rules_scala_scala_reflect/io_bazel_rules_scala_scala_reflect.stamp/scala-reflect-2.13.6-stamped.jar"
-        ),
-        "file://\$BAZEL_OUTPUT_BASE_PATH/execroot/__main__/bazel-out/k8-fastbuild/bin/scala_targets/library.jar"
-      )
+      ScalacOptionsItem(it, emptyList(), listOf("file://\$BAZEL_OUTPUT_BASE_PATH/execroot/__main__/bazel-out/k8-fastbuild/bin/external/io_bazel_rules_scala_scala_library/io_bazel_rules_scala_scala_library.stamp/scala-library-2.13.6-stamped.jar", "file://\$BAZEL_OUTPUT_BASE_PATH/execroot/__main__/bazel-out/k8-fastbuild/bin/external/io_bazel_rules_scala_scala_reflect/io_bazel_rules_scala_scala_reflect.stamp/scala-reflect-2.13.6-stamped.jar"), "file://\$BAZEL_OUTPUT_BASE_PATH/execroot/__main__/bazel-out/k8-fastbuild/bin/scala_targets/library.jar")
```
Comment (agluszak) [inline e2e/src/main/kotlin/org/jetbrains/bsp/bazel/BazelBspScalaProjectTest.kt:28]: nit: could you revert the formatting change?

**Review by agluszak — APPROVED** [review]

**Pull request merged**